autorest/client.go

```diff
@@ -72,6 +74,52 @@ const (
 `
 )
 
+type logLevelType int
+
+const (
+	logLevelNone logLevelType = iota
+	logLevelBasic
+	logLevelFull
+)
+
+const (
+	logTimestamp = "2006-01-02T15:04:05.0000000Z07:00"
+)
+
+var logLevel = logLevelNone
+var logFile *os.File
+
+func init() {
+	loadLogSettings()
+}
+
+// loading of log settings is placed in its own func to facilitate testing
+func loadLogSettings() {
+	// attempt to load logging level - this must be no fail
+	llStr := strings.ToLower(os.Getenv("AZURE_GO_AUTOREST_LOG_LEVEL"))
+	switch llStr {
+	case "basic":
+		logLevel = logLevelBasic
+	case "full":
+		logLevel = logLevelFull
+	}
+	if logLevel > logLevelNone {
+		// default to stderr
+		logFile = os.Stderr
+		lfStr := strings.ToLower(os.Getenv("AZURE_GO_AUTOREST_LOG_FILE"))
+		if lfStr == "stdout" {
```

Comment:
Given that this will need to run in case-sensitive environments, I'd refactor to this:

``` Go
lfStr := os.Getenv("AZURE_GO_AUTOREST_LOG_FILE")
if strings.EqualFold(lfStr, "stdout") {
    logFIle = os.Stdout
} else if lfStr != "" {
    // open/create file and truncate
}
```

Reply:
I do have a call to ```strings.ToLower``` to avoid case-sensitivity however I think ```strings.EqualFold``` is a better choice.

Reply:
What I was getting at is that using `strings.ToLower` means that the original casing provided by the author gets lost when you `os.Create` using `lfStr` later. :)

Reply:
Good point!